code/modules/admin/verbs/shiphealthpanelUI.dm

```diff
@@ -0,0 +1,20 @@
+GLOBAL_DATUM_INIT(ship_health_panel_UI, /datum/shiphealthpanelUI, new)
+
+/datum/shiphealthpanelUI/tgui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "shiphealthpanelUI")
+		ui.open()
+
+/datum/shiphealthpanelUI/ui_data(mob/user)
+	var/list/data = list()
+	data["missile"] = GLOB.ship_hits_tally["times_hit_missile"]
+	data["railgun"] = GLOB.ship_hits_tally["times_hit_railgun"]
+	data["odc"] = GLOB.ship_hits_tally["times_hit_odc"]
+	data["aaboiler"] = GLOB.ship_hits_tally["times_hit_aaboiler"]
```

Comment:
If you take the first suggestion then it would be like this:
```
	data["missile"] = GLOB.ship_hits_tally[SVS_WEAPON_MISSILE]
	data["railgun"] = GLOB.ship_hits_tally[SVS_WEAPON_RAILGUN]
	data["odc"] = GLOB.ship_hits_tally[SVS_WEAPON_ODC]
	data["aaboiler"] = GLOB.ship_hits_tally[SVS_WEAPON_AABOILER]
```